packages/dds/map/src/directory.ts

```diff
@@ -693,17 +698,34 @@ export class SharedDirectory extends SharedObject<ISharedDirectoryEvents> implem
         return this.localValueMaker.fromSerializable(serializable);
     }
 
+    private getParentDirectory(relativePath: string): IDirectory | undefined {
+        const absolutePath = this.makeAbsolute(relativePath);
+        const subdirs = absolutePath.substr(1).split(posix.sep);
+        if (subdirs.length === 0) {
+            return undefined;
+        }
+        subdirs.pop();
+        return this.getWorkingDirectory(subdirs.join(posix.sep));
+    }
+
+    private isSubDirectoryDeletePending(relativePath: string): boolean {
```

Comment:
Would be great to add some comments above functions. I assume this function tells if there is local pending (non-acked yet) delete, right?